aptos-move/framework/aptos-framework/sources/configs/stake.move

```diff
@@ -752,31 +738,19 @@ module aptos_framework::stake {
         pool_address: address,
         amount: u64,
         owner_cap: &OwnerCapability,
-    ) acquires StakePool, StakePoolEvents, ValidatorSet {
+    ) acquires StakePool, StakePoolEvents {
         assert!(owner_cap.pool_address == pool_address, error::invalid_argument(ENOT_OWNER));
 
         // Short-circuit if amount to unlock is 0 so we don't emit events.
         if (amount == 0) {
             return
         };
 
+        // Unlocked coins are moved to pending_inactive. When the current lockup cycle expires, they will be moved into
+        // inactive in the earliest possible epoch transition.
         let stake_pool = borrow_global_mut<StakePool>(pool_address);
         let unlocked_stake = coin::extract<AptosCoin>(&mut stake_pool.active, amount);
-
-        // Withdrawal is not allowed if lockup has not expired.
-        // This is also enforced even if the validator is not active in order to prevent a validator from intentionally
-        // leave the set early to be able to unlock.
-        if (stake_pool.locked_until_secs > timestamp::now_seconds()) {
-            abort error::invalid_argument(EWITHDRAW_NOT_ALLOWED)
-        };
-
-        // Move requested amount to pending_inactive if the validator is already active.
-        // Otherwise, we can move directly to inactive.
-        if (is_current_epoch_validator(pool_address)) {
-            coin::merge<AptosCoin>(&mut stake_pool.pending_inactive, unlocked_stake);
-        } else {
-            coin::merge<AptosCoin>(&mut stake_pool.inactive, unlocked_stake);
-        };
+        coin::merge<AptosCoin>(&mut stake_pool.pending_inactive, unlocked_stake);
```

Comment:
hmm, so pending_inactive only changes after lockup expires instead of at epoch boundary?

Reply:
Yes. This is the best option for a recurring lockup I think. Otherwise, users would need to time their unlock/withdraw to coincide with the lockup expiration.